skbio/metadata/_interval.py

```diff
@@ -90,6 +91,10 @@ def __init__(self, interval_metadata, locations,
 
     def _add(self):
         """Add the current ``Interval`` to the IntervalMetadata object."""
+        if self.locations[-1][-1] > self._interval_metadata.upper_bound:
+            raise ValueError('Cannot set `locations` (%r) with coordinate '
```

Comment:
Can you add unit tests for adding `Interval` objects that fall outside the `IntervalMetadata` range?